vm/src/obj/objint.rs

```diff
@@ -289,6 +291,11 @@ fn int_add(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     );
     if objtype::isinstance(other, &vm.ctx.int_type()) {
         Ok(vm.ctx.new_int(get_value(zelf) + get_value(other)))
+    } else if objtype::isinstance(other, &vm.ctx.complex_type()) {
```

Comment:
how acout call __radd__ on the complex instead? This way int object doesn't have to know about operations on complex numbers.

Reply:
Didn't know `about ``__radd__```! will do.

Reply:
(technically, CPython uses the same C function for both `1+1j` and `1j+1` - it promotes any side to complex if needed. However, I'd assume that since we can't do it this way now, `__radd__` would be closer to expected behavior)